core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -127,13 +126,6 @@ public void delete(PageId pageId) throws IOException, PageNotFoundException {
       throw new PageNotFoundException(p.toString());
     }
     Files.delete(p);
-    Path parent = Preconditions.checkNotNull(p.getParent(),
```

Comment:
removing parent is unnecessary and introduces race on the parent dir among threads, causing put failure unnecessarily. Remove this logic